ui/js/controllers/main.js

```diff
@@ -1,27 +1,63 @@
 "use strict";
 
 treeherderApp.controller('MainCtrl', [
-    '$scope', '$rootScope', '$routeParams', '$location', '$timeout',
+    '$scope', '$rootScope', '$routeParams', '$location', '$timeout', '$q',
     'ThLog', 'ThRepositoryModel', 'thPinboard', 'thTabs', '$document',
-    'thClassificationTypes', 'thEvents', '$interval', '$window',
+    'thClassificationTypes', 'thEvents', '$interval', '$window', 'thNotify',
     'ThExclusionProfileModel', 'thJobFilters', 'ThResultSetStore',
-    'thDefaultRepo', 'thJobNavSelectors', 'thTitleSuffixLimit',
+    'thDefaultRepo', 'thJobNavSelectors', 'thTitleSuffixLimit', '$http',
     function MainController(
-        $scope, $rootScope, $routeParams, $location, $timeout,
+        $scope, $rootScope, $routeParams, $location, $timeout, $q,
         ThLog, ThRepositoryModel, thPinboard, thTabs, $document,
-        thClassificationTypes, thEvents, $interval, $window,
+        thClassificationTypes, thEvents, $interval, $window, thNotify,
         ThExclusionProfileModel, thJobFilters, ThResultSetStore,
-        thDefaultRepo, thJobNavSelectors, thTitleSuffixLimit) {
+        thDefaultRepo, thJobNavSelectors, thTitleSuffixLimit, $http) {
         var $log = new ThLog("MainCtrl");
 
         // Query String param for selected job
         var QS_SELECTED_JOB = "selectedJob";
 
+        //var revisionPollInterval = 60000 * 5;
+        var revisionPollInterval = 15000;
+
         // Ensure user is available on initial page load
         $rootScope.user = {};
 
         thClassificationTypes.load();
 
+        $rootScope.checkServerRevision = function() {
+            return $q(function(resolve, reject) {
+                $http({
+                    method: 'GET',
+                    url: '/revision.txt'
+                }).then(function successCallback(response) {
+                    resolve(response.data);
+                }, function errorCallback(response) {
+                    reject("Error loading revision.txt: " + response.statusText);
+                });
+            });
+        };
+
+        if(_.isUndefined($rootScope.serverRev)) {
```

Comment:
I don't think we need the check for isUndefined (you can assume it will be when this code is executed)